hs_core/page_processors.py

```diff
@@ -119,10 +120,15 @@ def get_page_context(page, user, resource_edit=False, extended_metadata_layout=N
                                 tool_url, [content_model.get_hs_term_dict(), hs_term_dict_user])
                             if tool_url_new is not None:
                                 tl = {'title': str(tool_res_obj.metadata.title.value),
+                                      'tool_landing_page': "/resource/{tool_short_id}/".
+                                      format(tool_short_id=tool_res_obj.short_id),
                                       'icon_url': tool_icon_url,
                                       'url': tool_url_new}
                                 relevant_tools.append(tl)
 
+                                if tool_res_obj in get_my_resources_list(request) or tool_res_obj.approved:
```

Comment:
The following call is an expensive one:
```get_my_resources_list(request)```

We don't want to call this function repeatedly inside the for loop since the list of resources returned by this call does not change.